qlib/utils/time.py

```
@@ -88,25 +88,25 @@ def is_single_value(start_time, end_time, freq, region: str = REG_CN):
     if region == REG_CN:
-        if end_time - start_time < freq:
+        if end_time - start_time > freq:
             return True
         if start_time.hour == 11 and start_time.minute == 29 and start_time.second == 0:
-            return True
+            return False
         if start_time.hour == 14 and start_time.minute == 59 and start_time.second == 0:
-            return True
-        return False
+            return False
+        return True
```

Ruff (0.11.9)

Return the negated condition directly

Inline condition

(SIM103)